sklearn/base.py

```diff
@@ -378,7 +384,7 @@ def _check_n_features(self, X, reset):
                                        self.n_features_in_)
                 )
 
-    def _validate_data(self, X, y=None, reset=True,
+    def _validate_data(self, X, y=__NO_Y, reset=True,
```

Comment:
I would just set `y="no_validation"` instead of defining a module level constant as default placeholder with a rather implicit name.

Reply:
That would make the docstring more readable.